lib/dnsmadeeasy-rest-api.rb

```diff
@@ -59,15 +59,15 @@ def records_for(domain_name)
   end
 
   def find(domain_name, name, type)
-    records = records_for(domain_name)
-    records['data'].detect { |r| r['name'] == name && r['type'] == type }
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
+
+    records['data'].first
   end
 
   def find_record_id(domain_name, name, type)
-    records = records_for(domain_name)
+    records = get "/dns/managed/#{get_id_by_domain(domain_name)}/records?recordName=#{name}&type=#{type}"
```

Comment:
updated in https://github.com/3dna/dnsmadeeasy-rest-api/pull/2/commits/95f58fecb29d6c25633c7b17bad224edfe14f8d8 to make it also work for records_for